plugins/module_utils/network/nxos/facts/fc_interfaces/fc_interfaces.py

```diff
@@ -0,0 +1,103 @@
+# -*- coding: utf-8 -*-
+# Copyright 2023 Red Hat
+# GNU General Public License v3.0+
+# (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+
+import operator
+import re
+
+from ansible_collections.ansible.netcommon.plugins.module_utils.network.common import utils
+
+from ansible_collections.cisco.nxos.plugins.module_utils.network.nxos.argspec.fc_interfaces.fc_interfaces import (
+    Fc_interfacesArgs,
+)
+from ansible_collections.cisco.nxos.plugins.module_utils.network.nxos.rm_templates.fc_interfaces import (
+    Fc_interfacesTemplate,
+)
+
+
+__metaclass__ = type
+
+"""
+The nxos fc_interfaces fact class
+It is in this file the configuration is collected from the device
+for a given resource, parsed, and the facts tree is populated
+based on the configuration.
+"""
+
+
+class Fc_interfacesFacts(object):
+    """The nxos fc_interfaces facts class"""
+
+    def __init__(self, module, subspec="config", options="options"):
+        self._module = module
+        self.argument_spec = Fc_interfacesArgs.argument_spec
+
+    def get_interfaces_data(self, connection):
+        return connection.get("show running-config interface all")
```

Comment:
One standard convention that we used in all modules was to _not_ include default values while pulling running-config. This let's us ignore those values (while rendering facts) that do not show up in `show running-config | section interface`.

Can you share we are doing this here?

Reply:
This also makes the `deleted` operation behave differently than all other interfaces modules.

Reply:
if any config are default then it wont show up in show running-config  hence we use show running-config interface all to get the default values

Reply:
deleted for interfaces module means moving to default values isnt it?

Reply:
> deleted for interfaces module means moving to default values isnt it?

Yes, but we do not consider the defaults that show up in `running-config all`? Did you have a look at how it is done in the nxos_interfaces module?